web/modules/custom/farm_nfa/js/farmOS.map.behaviors.farm_nfa_gfw_layers.js

```diff
@@ -2,13 +2,83 @@
-        farmNfaPlotGfwApiMap(instance,'deforestation', 'https://data-api.globalforestwatch.org/dataset/gfw_integrated_alerts/v20230215/query/json');
+        const fireAlertsUrl = 'https://data-api.globalforestwatch.org/dataset/nasa_viirs_fire_alerts/v20220726/query/json';
+        const deforestationAlertsUrl = 'https://data-api.globalforestwatch.org/dataset/gfw_integrated_alerts/v20230215/query/json';
+        farmNfaPlotGfwApiMap(instance, 'fire', fireAlertsUrl);
+        farmNfaPlotGfwApiMap(instance,'deforestation', deforestationAlertsUrl);
+        const dateApplyButton = document.querySelector('.comiseo-daterangepicker-buttonpanel button');
+        const dateUpdateButtons = document.querySelectorAll('[role="menuitem"]');
+        dateUpdateButtons.forEach((dateUpdateButton) => {
+          dateUpdateButton.addEventListener('click', () => {
+            updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl);
+          });
+        });
+        dateApplyButton.addEventListener('click', () => {
+          updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl);
+        });
       }
     }
 }())
 
-async function farmNfaPlotGfwApiMap(instance, mapType, gfwApiUrl) {
+async function updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl) {
+  const dateRange = await getStartEndDate();
+  const map = instance.map;
+  const layers = map.getLayers().getArray();
+  const noOfLayers = layers.length;
+  for (let i = 2; i < noOfLayers; i++) {
+    await map.removeLayer(layers[layers.length - 1]);    
+  }
+  farmNfaPlotGfwApiMap(instance, 'fire', fireAlertsUrl, dateRange);
+  farmNfaPlotGfwApiMap(instance, 'deforestation', deforestationAlertsUrl, dateRange);
+}
+
+function getStartEndDate() {
+  return new Promise((resolve, reject) => { 
+    setTimeout(() => {
+      const dateRange = document.querySelector('.daterangepicker-container')?.innerText;
+      const dateRangeArray = dateRange?.split(' - ');
+      let startDate = dateRangeArray[0]?.trim();
+      let endDate = dateRangeArray[1]?.trim();
+      startDate = startDate && new Date(startDate);
+      endDate = endDate && new Date(endDate);
+      startDate = startDate && startDate.toISOString().split('T')[0];
+      endDate = endDate && endDate.toISOString().split('T')[0];
+      resolve({startDate, endDate});
+    }, 100);
+  });
+}
+
+// make a function to get the date of last 6 months
+function getLastThreeMonthsDate() {
+  const currentDate = new Date(); // Get current date
+  const year = currentDate.getFullYear();
+  const month = currentDate.getMonth();
+  const day = currentDate.getDate();
+  // Subtract 3 months from the current date
+  const lastSixMonthsDate = new Date(year, month - 3, day);
+  // Format the date as "YYYY-MM-DD"
+  const formattedDate = lastSixMonthsDate.toISOString().slice(0, 10);
+  return formattedDate;
+}
+
+async function farmNfaPlotGfwApiMap(instance, mapType, gfwApiUrl, dateRange) {
```

Comment:
I'm thinking that a lot of this can be done serverside if we sub element the date range picker element and add the defaults there.
Or maybe just a data property on the element.

Reply:
i see @pcambra so you are suggesting to make the api calls for data layers from the server side right ? if so how will we handle the api calls when the user updates the date range ? as i can't call the api as that logic would get segregated to the server side ?

Reply:
I'm not proposing to move the api calls to the server, my feedback is regarding the separation of concerns, here you're configuring and setting up the query, running it and rendering it in a layer.
My suggestion would be to move as much configuration as possible to the render array in Drupal, i.e. field names, country codes, date formats, so javascript is only implementing in an abstract way all that data.